### DISCUSSION TITLE
Why squid is removed from the repo?

### DISCUSSION THREAD

**Discussion by nnWhisperer** [GoogleCloudPlatform/cloud-foundation-fabric#2655] [Q&A] [answered]
Hello,

1. Looking at the blueprints, I saw about squid. [It](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/0d486fb34e64c368f8a4cb8c58092486c2440b7c/blueprints/networking/README.md) seems like a good solution to limit egress calls to other domains. I wonder, why have this been moved to a folder called __need_fixing and commented out?
2. If the reason is that google provides an alternative service for filtering to squid proxy and saw it as a rival service, then, what is that service we can use instead of squid?
![squid 1](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/0d486fb34e64c368f8a4cb8c58092486c2440b7c/blueprints/networking/filtering-proxy/squid.png?raw=true)

**Comment by wiktorn** [comment, marked as answer]
ad 1. It was moved to _need_fixing here #1936, and the rationale was that the repository was no longer available. Later is was removed in #2131
ad 2. You can now use [Secure Web Proxy](https://cloud.google.com/secure-web-proxy/docs/overview) as an alternative to Squid proxy. There is also a module to help you deploy it - [modules/net-swp](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/modules/net-swp)

**Reply by nnWhisperer** [reply]
Which repository was no longer available?

**Reply by nnWhisperer** [reply]
@wiktorn , thank you for your response, could you tell me that which repository was no longer available? I see that [squid repository still exists in github](https://github.com/squid-cache/squid)? Thank you in advance.

**Reply by wiktorn** [reply]
Container Registry repository that was used to publish the Docker image `gcr.io/pso-cft-fabric` is no longer available, and there are no plans to expose container images as a part of Fabric FAST project.

**Reply by nnWhisperer** [reply]
Not to expose the container images as part of the fabric, but I guess adding a code to use the cloud build trigger module to generate the image from the Dockerfile might have been another solution though. I guess the main reason is about the presence of the secure web proxy in gcp, starting with the removal of the image hosted on the gcr.io in the first place and then removing the feature here. Thank you.
